### PR TITLE
fix #38

### LINKED ISSUE #38: Please add support for PRODUCT
LeadPoint is currently sending  a field called product. This contains information regarding the loan type. It has no allowable values that I can spot. Recent leads are passing only the value REFINANCE.

Can we please have this added to the LeadPoint integration. Clients are unsure of what type of loan they are being sent from the vendor.

Sample lead:
https://next.leadconduit.com/events/55d492c675daefe917c20613?source_id=53d7ba989869170001000002&type=source

Please prioritize this @alexj as this is a customer request.

### PR DESCRIPTION
**Pull request by jgrayson676**
#38

### CODE REVIEW

File: src/inbound.coffee [hunk 1]
```diff
@@ -339,6 +343,11 @@ parseDate = (value) ->
   return unless value?
   moment(value)
 
+parsePurpose = (value) ->
+  return unless value?
+  return 'purchase' if value.toLowerCase().indexOf('newhome') > -1
```
Comment (alexkwolfe) [inline src/inbound.coffee]: Will the product always be specified? I'm worried about not checking for null or undefined here.

File: src/inbound.coffee [hunk 2]
```diff
@@ -130,6 +133,7 @@ request.variables = ->
     { name: 'mortgage.loan.amortization',                   type: 'string',  description: 'Loan amortization type: adjustable, fixed, or other' },
     { name: 'mortgage.loan.ltv',                            type: 'number',  description: 'Loan to value ratio (80.0005)' },
     { name: 'mortgage.loan.cash_out',                       type: 'number',  description: 'How much additional cash do you wish to borrow?' },
+    { name: 'mortgage.loan.purpose',                        type: 'string',  description: 'The loan\'s purpose. Either \'purchase\', \'refi\', or \'other\''}
```
Comment (cgrayson) [inline src/inbound.coffee]: [According to `leadconduit-fields`](https://github.com/activeprospect/leadconduit-fields/blob/master/fields/mortgage.json#L15), the 3rd possibility is 'equity'. You might check other inbound mortgage integrations, like LowerMyBills or Military.com to confirm that.

File: spec/inbound_spec.coffee [hunk 3]
```diff
@@ -178,6 +178,11 @@ describe 'Inbound', ->
       it 'should parse cash out amount', ->
         assert.equal integration.request(req(body(add_cash: 10000))).mortgage.loan.cash_out, '10000'
 
+      it 'should parse loan purpose', ->
+        assert.equal integration.request(req("PRODUCT=BP_REFI")).mortgage.loan.purpose, 'refi'
+        assert.equal integration.request(req("PRODUCT=BP_VOICE_NEWHOME")).mortgage.loan.purpose, 'purchase'
+        assert.equal integration.request(req("PRODUCT=SOMETHING_ELSE")).mortgage.loan.purpose, 'other'
+
```
Comment (cgrayson) [inline spec/inbound_spec.coffee]: Could use a test to verify `product` is set, too.

File: src/inbound.coffee [hunk 4]
```diff
@@ -339,6 +343,11 @@ parseDate = (value) ->
   return unless value?
   moment(value)
 
+parsePurpose = (value) ->
+  return unless value?
+  return 'purchase' if value.toLowerCase().indexOf('newhome') > -1
+  return 'refi' if value.toLowerCase().indexOf('refi') > -1
+  return 'other'
```
Comment (cgrayson) [inline src/inbound.coffee]: Not a bad idea to cover the default case, but I think for this field it should be one of the 3 values, or nothing (`null`).

File: src/inbound.coffee (hunk 1 above)
Comment (cgrayson) [inline src/inbound.coffee]: Doesn't line 347 cover that, or am I misunderstanding you?

Comment (alexkwolfe) [inline src/inbound.coffee]: :hammer: :hourglass: 

You're right. Dur.

**Comment by cgrayson** [comment]
LGTM.

**Pull request merged**